pandas/tests/window/moments/conftest.py

```diff
@@ -18,3 +18,51 @@ def binary_ew_data():
 @pytest.fixture(params=[0, 1, 2])
 def min_periods(request):
     return request.param
+
+
+base_functions_list = [
+    (lambda v: Series(v).count(), None, "count"),
+    (lambda v: Series(v).max(), None, "max"),
+    (lambda v: Series(v).min(), None, "min"),
+    (lambda v: Series(v).sum(), None, "sum"),
+    (lambda v: Series(v).mean(), None, "mean"),
+    (lambda v: Series(v).std(), 1, "std"),
+    (lambda v: Series(v).cov(Series(v)), None, "cov"),
+    (lambda v: Series(v).corr(Series(v)), None, "corr"),
+    (lambda v: Series(v).var(), 1, "var"),
+    # restore once GH 8086 is fixed
+    # lambda v: Series(v).skew(), 3, 'skew'),
+    # (lambda v: Series(v).kurt(), 4, 'kurt'),
+    # restore once GH 8084 is fixed
+    # lambda v: Series(v).quantile(0.3), None, 'quantile'),
+    (lambda v: Series(v).median(), None, "median"),
+    (np.nanmax, 1, "max"),
+    (np.nanmin, 1, "min"),
+    (np.nansum, 1, "sum"),
+    (np.nanmean, 1, "mean"),
+    (lambda v: np.nanstd(v, ddof=1), 1, "std"),
+    (lambda v: np.nanvar(v, ddof=1), 1, "var"),
+    (np.nanmedian, 1, "median"),
+]
+
+no_nan_functions_list = [
+    (np.max, None, "max"),
+    (np.min, None, "min"),
+    (np.sum, None, "sum"),
+    (np.mean, None, "mean"),
+    (lambda v: np.std(v, ddof=1), 1, "std"),
+    (lambda v: np.var(v, ddof=1), 1, "var"),
+    (np.median, None, "median"),
+]
+
+
+@pytest.fixture(scope="session")
+def base_functions():
+    """Fixture for base functions."""
+    return base_functions_list
```

Comment:
can you make these return single functions at a time? e.g. use `pytest.parametrize`? or is that big to actually re-write?

Reply:
yeah, i tried to do this so that each will be returned at a time, and thus also avoid the inner loop in the actual test methods.

but there are two issues:
1. we already have quite large amount of tests (hundreds where it is used, execution time is around 25s) due to parametrization, and the number of tests using this fixture will increase the tests by 20x. So need to take a lot of more time to finish. And also `base_functions` and `no_nan_functions` will be determined based on the value of another parametrization fixture, so also needs to make split on this.
2. a more rabbit-hole-alike issue is the warning message, for some of the combinations bewteen: `data` and `functions`, there will be warnings raised because data is empty, and slicing will raise warnings for some functions. Right now, since they are as a whole (by the inner loop), we are certain it will raise this message in one of those hundreds of combinations. But if we parametrize this, then we also need to separate these to address which will raise. and i am not sure if it is worth doing it.

any thoughts/comments? @jreback

Reply:
ok that's fine, leave as is then